modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -148,14 +133,6 @@ if (CONFIG_BUILD_WITH_TFM)
     list(APPEND TFM_CMAKE_ARGS -D${partition}=${val})
   endforeach()
 
-  # Enable TFM crypto modules as specified in Kconfig
-  foreach(module ${TFM_CRYPTO_MODULES})
-    if (CONFIG_TFM_${module}_ENABLED)
```

Comment:
I believe this maps the following Kconfigs from Zephyr to the equivalent TF-M settings.

Zephyr settings are found here:
https://github.com/zephyrproject-rtos/zephyr/blob/7a21c6f16aa0bffa93975841d5e1890ca6d0705f/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules#L10
https://github.com/zephyrproject-rtos/zephyr/blob/7a21c6f16aa0bffa93975841d5e1890ca6d0705f/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules#L17
https://github.com/zephyrproject-rtos/zephyr/blob/7a21c6f16aa0bffa93975841d5e1890ca6d0705f/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules#L25
https://github.com/zephyrproject-rtos/zephyr/blob/7a21c6f16aa0bffa93975841d5e1890ca6d0705f/modules/trusted-firmware-m/Kconfig.tfm.crypto_modules#L33
etc.

Reply:
This only proves that the dead code reads other code.

I cannot find any place in the TF-M build system that reads CMake variables like:

CRYPTO_RNG_MODULE_ENABLED

Nor can I prove a negative :(

Reply:
I have not not checked if the Kconfig is dead so I'd prefer not to delete it.

But if you believe it is dead I would be happy to delete it as well.

Reply:
@tejlmand : CMake is also reporting that it is dead code:

CMake Warning:
  Manually-specified variables were not used by the project:

    CRYPTO_RNG_MODULE_ENABLED

Reply:
> I cannot find any place in the TF-M build system that reads CMake variables like:
> 
> CRYPTO_RNG_MODULE_ENABLED

It's described here:
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/0b898c9b72171b0a260c0bb64a92ea4713f9e931/docs/configuration/index.rst?plain=1#L129

and used here:
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/0b898c9b72171b0a260c0bb64a92ea4713f9e931/config/config_base.h#L51-L54
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/0b898c9b72171b0a260c0bb64a92ea4713f9e931/secure_fw/partitions/crypto/crypto_rng.c#L27

But I also know that TF-M has started to use / support Kconfig as configuration system, but i'm not up-to-date on the Zephyr Kconfig --> TF-M Kconfig integration work. I assume you has more knowledge on this part.
If the TF-M Kconfig is now mapped into Zephyr Kconfig in an updated fashion, then it's likely that the old code (CMake + Kconfig) can be removed.

Ref: 
https://tf-m-user-guide.trustedfirmware.org/configuration/kconfig_system.html#the-kconfig-system
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/0b898c9b72171b0a260c0bb64a92ea4713f9e931/secure_fw/partitions/crypto/Kconfig.comp#L38-L39

Reply:
The usages you show are C preprocessor macros.

This dead code is writing to CMake variables that (again, can't prove a negative), I believe are unused.

There is no integration between Zephyr's Kconfig and TF-M's Kconfig.

I believe this CMake died when they moved from CMake-based configuration to C-based configuration.